src/ui/public/config/config.js

```diff
@@ -129,34 +35,84 @@ module.service('config', function (Private, kbnVersion, kbnIndex, $rootScope, bu
    *                             be stored. Defaults to the config key
    * @return {function} - an unbind function
    */
-  config.$bind = function ($scope, key, property) {
-    if (!property) property = key;
-
-    let update = function () {
-      $scope[property] = config.get(key);
-    };
+  config.$bind = function (scope, key, property = key) {
+    config.on(key, update, scope);
+    function update(value) {
+      scope[property] = value;
+    }
+  };
 
+  config.on = function (key, fn, scope = $rootScope) {
+    const postfix = key === `*` ? `` : `.${key}`;
```

Comment:
Out of curiosity, is the key passed here something we can validate? Not necessary, but if you think it'd be easy and reliable I think it would be handy